notification/logic/message/PostMessageLogic.java

```diff
@@ -0,0 +1,313 @@
+package com.open.jp.notification.logic.message;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+import javax.ws.rs.core.Response;
+
+import com.open.jp.notification.common.DataSourceHolder;
+import com.open.jp.notification.errorresponse.ErrorResponse;
+import com.open.jp.notification.logic.ErrorResponseEnum;
+import com.open.jp.notification.logic.PostLogicSendFcm;
+import com.open.jp.notification.logic.PostLogicSendFcmSetBody;
+import com.open.jp.notification.request.message.RequestMessage;
+import com.open.jp.notification.request.message.RequestMessageBody;
+import com.open.jp.notification.response.ResponseMessage;
+import com.open.jp.notification.response.ResponseMessageBodyIDList;
+import com.open.jp.notification.response.ResponseMessageBodyMessage;
+import com.open.jp.notification.util.StringUtil;
+
+
+@RequestScoped
+public class PostMessageLogic {
+  
+  @Inject
+  private DataSourceHolder ds;
+  
+  private StringUtil stringutil = new StringUtil();
+  
+  public PostMessageLogic() {
+      
+  }
+  
+  public  Response GenerateErrorRes(ErrorResponseEnum errorresponseenum) {
+      ErrorResponse erroresponse = new ErrorResponse();
+      erroresponse.setTime(LocalDateTime.now().toString()); 
+      erroresponse.setCode(errorresponseenum.getErrorStatusCodeMessage());
+      erroresponse.setMessage(errorresponseenum.getErrorMessage());
+      return Response.status(errorresponseenum.getErrorStatus()).entity(erroresponse).build();
+  }
+  
+  public Response service(RequestMessage request, RequestMessageBody body) {
+    
+    StringBuilder seqMID = new StringBuilder();
+    ResponseMessage response = new ResponseMessage();
+    StringBuilder token = new StringBuilder();
+    
+    List<ResponseMessageBodyIDList> idList = new  ArrayList<>();
+    Response IdListCheckResponse = IdListCheck(body);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *IdListCheckResponse*. The resource is returned at line 62. There are other execution paths that don&#39;t close the resource or return it, for example, when an else condition at line 61 evaluates to true. Ensure *IdListCheckResponse* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).